lib/ansible/modules/packaging/os/package_facts.py

```diff
@@ -255,8 +261,11 @@ def is_available(self):
         we_have_lib = super(APT, self).is_available()
         if not we_have_lib:
             for exe in ('apt', 'apt-get', 'aptitude'):
-                if get_bin_path(exe):
-                    module.warn('Found "%s" but %s' % (exe, missing_required_lib('apt')))
+                try:
+                    if get_bin_path(exe):
+                        module.warn('Found "%s" but %s' % (exe, missing_required_lib('apt')))
+                        break
+                except ValueError:
```

Comment:
I think the ValueError should continue rather than break.  The logic below is what I think is correct (and I also took the opportunity to get rid of the conditional:

``` python
for exe in ('apt', 'apt-get', 'aptitude'):
    try:
        get_bin_path(exe)
        module.warn('Found "%s" but %s' % (exe, missing_required_lib('apt')))
        break
    except ValueError:
        continue
```
I'm not sure if removing the conditional altogether reads clearer than the original.  I might do it this way as a compromise:

``` python
for exe in ('apt', 'apt-get', 'aptitude'):
    try:
        get_bin_path(exe)
    except ValueError:
        continue
    else:
        module.warn('Found "%s" but %s' % (exe, missing_required_lib('apt')))
        break
```

Reply:
Going with the second suggestion because I think it reads better.